cluster_management/src/main/java/com/pinterest/rocksplicator/OnlineOfflineStateModelFactory.java

```diff
@@ -161,24 +161,24 @@ public void onBecomeOfflineFromError(Message message, NotificationContext contex
       Utils.logTransitionMessage(message);
     }
 
-    private Admin.Client getLocalAdminClient() throws TTransportException {
-      TSocket sock = new TSocket("localhost", adminPort);
-      sock.open();
-      return new Admin.Client(new TBinaryProtocol(sock));
-    }
-
-    private void checkSanity(String fromState, String toState, Message message) {
+    public void checkSanity(String fromState, String toState, Message message) {
       if (fromState.equalsIgnoreCase(message.getFromState())
           && toState.equalsIgnoreCase(message.getToState())
           && resourceName.equalsIgnoreCase(message.getResourceName())
           && partitionName.equalsIgnoreCase(message.getPartitionName())) {
         return;
       }
 
-      LOG.error("Invalid meesage: " + message.toString());
+      LOG.error("Invalid message: " + message.toString());
```

Comment:
could you move getLocalAdminClient() back to it original place to minimize the number of changed lines? That will make it easier for people to review.

It took me a while to figure out that the real changes are this typo and private->public.

Can you change checkSanity to protected.

Reply:
Is it good java practice to have private functions above public functions?